### ISSUE TITLE
[Xtext-Generator]JavaFileAccess creates unnecessary imports

### ISSUE THREAD

**Issue by cdietrich** [eclipse/xtext-core#269]
Consider the following Grammar

```
Model:
	test=WarnTest
;
enum WarnTest:
	A = 'a' |
	B = 'b'
;
```

this gives the following unnecessary import in the generated Grammar Access

```
import org.eclipse.xtext.service.AbstractElementFinder.AbstractEnumRuleElementFinder; // this one is not needed
import org.eclipse.xtext.service.AbstractElementFinder.AbstractGrammarElementFinder;
```

**Comment by cdietrich** [comment]
we already have the `JavaFileAccess.DONT_IMPORT_NESTED_TYPES` feature. maybe applying it to the grammar access solves the problem

**Comment by cdietrich** [comment]
@meysholdt do you have any comments on this issue?

**Comment by miklossy** [comment]
I also encountered this problem recently:
![screenshot](https://user-images.githubusercontent.com/20393472/69157584-70887100-0ae5-11ea-8256-e31d9ed4affa.png)

**Comment by cdietrich** [comment]
@miklossy are you going to look into this issue?

**Comment by miklossy** [comment]
Also reproducible with the Eclipse GEF DOT [DotArrowType.xtext](https://github.com/eclipse/gef/blob/master/org.eclipse.gef.dot/src/org/eclipse/gef/dot/internal/language/DotArrowType.xtext) grammar:
![image](https://user-images.githubusercontent.com/20393472/72611586-c7b29280-392a-11ea-851b-adf6bab04d95.png)

**Comment by miklossy** [comment]
Currently, the following 4 Xtext grammars of the [Eclipse GEF DOT](https://github.com/eclipse/gef/wiki/DOT-User-Guide) project are affected by this problem:
![screenshot](https://user-images.githubusercontent.com/20393472/73069728-c6dfaa80-3eae-11ea-9bf4-507ac7a5bdf5.png)

**Comment by miklossy** [comment]
I also encountered this problem on the [TargetPlatform-DSL](https://github.com/eclipse-cbi/targetplatform-dsl) project.
![screenshot](https://user-images.githubusercontent.com/20393472/80958364-34cc3400-8e05-11ea-97c3-695438b307bb.png)

**Comment by miklossy** [comment]
Fixed in 2.22
